### PR TITLE
fix(relay): rust build fails after tokio bump from 0.2.25 to 1.18.5

### LINKED ISSUE #2349: fix(relay): rust build fails after tokio bump from 0.2.25 to 1.18.5
**Describe the bug**

Since the update of tokio version to 1.x, the build of relay is failing.

**To Reproduce**

Go to `weaver/core/relay` and run `make build-local`.

**Expected behavior**

Build should succeed.

**Logs/Stack traces**

```
   Compiling relay v1.5.4 (/opt/relay)
error: The #[tokio::main] macro requires rt or rt-multi-thread.
 --> src/client_tls.rs:8:1
  |
8 | #[tokio::main]
  | ^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `tokio::main` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: unused imports: `NetworkQuery`, `network_client::NetworkClient`
 --> src/client_tls.rs:2:36
  |
2 | use weaverpb::networks::networks::{network_client::NetworkClient, NetworkQuery};
  |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `std::net::SocketAddr`
  --> driver/driver.rs:14:5
   |
14 | use std::net::SocketAddr;
   |     ^^^^^^^^^^^^^^^^^^^^

warning: unused import: `std::net::ToSocketAddrs`
  --> driver/driver.rs:15:5
   |
15 | use std::net::ToSocketAddrs;
   |     ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `tonic::transport::Server`
  --> driver/driver.rs:19:5
   |
19 | use tonic::transport::Server;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `dummy_driver`
   --> driver/driver.rs:246:2
    |
246 | }
    |  ^ consider adding a `main` function to `driver/driver.rs`

Some errors have detailed explanations: E0425, E0432, E0601, E0603.
For more information about an error, try `rustc --explain E0425`.
warning: `relay` (bin "dummy-driver") generated 5 warnings
error: could not compile `relay` due to 6 previous errors; 5 warnings emitted
The command '/bin/sh -c cd /opt/relay && RUSTFLAGS="-Clinker=musl-gcc -Crelocation-model=static" cargo build --release --target x86_64-unknown-linux-musl' returned a non-zero code: 101
make: *** [Makefile:90: build-server-local] Error 101
```

@petermetz Since you are familiar with Rust, please let us know if you could provide any insight on how we can update the codebase to adapt to newer tokio version.

### PR DESCRIPTION
**Pull request by sandeepnRES**
Additionally:
    - upgrade other dependencies for relay
    - added tls based unit test in relay

Closes #2349

### CODE REVIEW

**Review by petermetz — APPROVED** [review]
@sandeepnRES LGTM, thank you!

**Review by takeutak — APPROVED** [review]

File: weaver/core/relay/driver/driver.rs [hunk 1]
```diff
@@ -34,54 +36,57 @@ pub struct URI {
...
+                    send_driver_mock_state_helper(client, request_id.to_string());
+                }
+                Err(e) => {
+                    // TODO: Add better error handling (Attempt a few times?)
+                    panic!("Failed to connect to client. Error: {}", e.to_string());
+                }
             }
         }
-
+
```
Comment (VRamakrishna) [inline weaver/core/relay/driver/driver.rs:89]: Unnecessary indent

File: weaver/core/relay/driver/driver.rs [hunk 2]
```diff
@@ -96,42 +101,51 @@ impl DriverCommunication for DriverCommunicationService {
...
-            .config_lock
-            .read()
-            .await
-            .get_str("hostname")
-            .expect("Hostname table does not exist");
+            .get_table("relays")
+            .expect("No relays table in config file");
+        let relay_uri = relays_table
+            .get("Dummy_Relay")
+            .expect("No Dummy_Relay in config file relays table");
```
Comment (VRamakrishna) [inline weaver/core/relay/driver/driver.rs:112]: Same comment as earlier.

File: weaver/core/relay/driver/driver.rs (hunk 1 above)
Comment (sandeepnRES) [reply weaver/core/relay/driver/driver.rs]: Fixed

File: weaver/core/relay/driver/driver.rs [hunk 3]
```diff
@@ -34,54 +36,57 @@ pub struct URI {
...
             .config_lock
             .read()
             .await
-            .get_str("hostname")
-            .expect("Hostname table does not exist");
+            .get_table("relays")
+            .expect("No relays table in config file");
+        let relay_uri = relays_table
+            .get("Dummy_Relay")
+            .expect("No Dummy_Relay in config file relays table");
```
[the remark this answers is not in the document]
Comment (sandeepnRES) [reply weaver/core/relay/driver/driver.rs]: Made it general by taking it from query recieved.

**Review by VRamakrishna — APPROVED** [review]

**Pull request closed without merging**